### PR TITLE
MON-2988: fix myAccount special characters update

### PR DESCRIPTION
**Pull request by garnier-quentin**
## Description

Fix when you have special characters in your name. for example: é

## Type of change

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [X] 19.10.x
- [X] 20.04.x
- [X] 20.10.x
- [X] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Create an user with special character (é for example). Connect with that user and edit it (save). You cannot login. 

## Checklist

- [X] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by jeremyjaouen — DISMISSED** [review]

File: www/include/Administration/myAccount/DB-Func.php [hunk 1]
```diff
@@ -126,66 +126,68 @@ function updateContact($contact_id = null)
...
         if ($encryptType == 1) {
-            $rq .= "contact_passwd = '" . md5($ret["contact_passwd"]) . "', ";
+            $password_encrypted = md5($ret['contact_passwd']);
         } elseif ($encryptType == 2) {
-            $rq .= "contact_passwd = '" . sha1($ret["contact_passwd"]) . "', ";
+            $password_encrypted = sha1($ret['contact_passwd']);
         } else {
-            $rq .= "contact_passwd = '" . md5($ret["contact_passwd"]) . "', ";
+            $password_encrypted = md5($ret['contact_passwd']);
         }
```
Comment (jeremyjaouen) [inline www/include/Administration/myAccount/DB-Func.php:148]: I think this part could be simplified:

if ($encryptType == 2) {
    $password_encrypted = sha1($ret['contact_passwd']);
} else {
    $password_encrypted = md5($ret['contact_passwd']);
}

**Review by jeremyjaouen — DISMISSED** [review]

File: www/include/Administration/myAccount/DB-Func.php [hunk 2]
```diff
@@ -126,66 +126,68 @@ function updateContact($contact_id = null)
...
+          'contact_alias = :contactAlias, ' .
+          'contact_location = :contactLocation, ' .
+          'contact_lang = :contactLang, ' .
+          'contact_email = :contactEmail, ' .
+          'contact_pager = :contactPager, ' .
+          'default_page = :defaultPage, ' .
+          'contact_js_effects = :contactJsEffects, ' .
+          'contact_autologin_key = :contactAutologinKey';
+    $password_encrypted = null;
+    if (isset($ret['contact_passwd']) && $ret['contact_passwd']) {
```
Comment (kduret) [inline www/include/Administration/myAccount/DB-Func.php:140]:
```suggestion
    if (!empty($ret['contact_passwd'])) {
```

**Review by kduret — DISMISSED** [review]

File: www/include/Administration/myAccount/DB-Func.php [hunk 3]
```diff
@@ -126,66 +126,66 @@ function updateContact($contact_id = null)
...
-    $rq .= "WHERE contact_id = :contactId";
+    $rq .= ' WHERE contact_id = :contactId';
 
     $stmt = $pearDB->prepare($rq);
+    $stmt->bindValue(':contactName', $ret['contact_name'], \PDO::PARAM_STR);
+    $stmt->bindValue(':contactAlias', $ret['contact_alias'], \PDO::PARAM_STR);
+    $stmt->bindValue(':contactLang', $ret['contact_lang'], \PDO::PARAM_STR);
+    $stmt->bindValue(
+        ':contactEmail',
+        !empty($ret['contact_email']) ? $ret['contact_email'] : null,
```
Comment (sc979) [inline www/include/Administration/myAccount/DB-Func.php:157]: htmlentities is required

**Review by sc979 — CHANGES_REQUESTED** [review]

File: www/include/Administration/myAccount/DB-Func.php [hunk 4]
```diff
@@ -125,67 +125,83 @@ function updateContact($contact_id = null)
...
-        : $rq .= "NULL, ";
-
-    if (isset($ret["contact_passwd"]) && $ret["contact_passwd"]) {
-        if ($encryptType == 1) {
-            $rq .= "contact_passwd = '" . md5($ret["contact_passwd"]) . "', ";
-        } elseif ($encryptType == 2) {
-            $rq .= "contact_passwd = '" . sha1($ret["contact_passwd"]) . "', ";
+    $ret['contact_email'] = CentreonUtils::escapeSecure(
+        !empty($ret['contact_email']) ? $ret['contact_email'] : '',
+        CentreonUtils::ESCAPE_ILLEGAL_CHARS
```
Comment (sc979) [inline www/include/Administration/myAccount/DB-Func.php:130]:
```suggestion
    $ret['contact_email'] = !empty($ret['contact_email'])
        ? CentreonUtils::escapeSecure($ret['contact_email'], CentreonUtils::ESCAPE_ILLEGAL_CHARS)
        : '';
        
```

File: www/include/Administration/myAccount/DB-Func.php [hunk 5]
```diff
@@ -125,67 +125,83 @@ function updateContact($contact_id = null)
...
+        !empty($ret['contact_pager']) ? $ret['contact_pager'] : '',
+        CentreonUtils::ESCAPE_ILLEGAL_CHARS
+    );
+    $ret['contact_autologin_key'] = CentreonUtils::escapeSecure(
+        !empty($ret['contact_autologin_key']) ? $ret['contact_autologin_key'] : '',
+        CentreonUtils::ESCAPE_ILLEGAL_CHARS
+    );
+    $ret['contact_lang'] = CentreonUtils::escapeSecure(
+        !empty($ret['contact_lang']) ? $ret['contact_lang'] : '',
+        CentreonUtils::ESCAPE_ILLEGAL_CHARS
```
Comment (sc979) [inline www/include/Administration/myAccount/DB-Func.php:142]: kindly, could you modify as it's suggested upper ?

**Review by sc979 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**